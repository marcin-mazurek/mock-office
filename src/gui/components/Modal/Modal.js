import React from 'react';
import PropTypes from 'prop-types';

const createModal = (types) => {
  const Modal = ({ component, onOverlayClick }) => {
    const ModalComponent = types[component];
    const isOpen = !!ModalComponent;

    return isOpen
      ? (
        <div
          className="modal"
        >
          <button className="modal__overlay" onClick={onOverlayClick} />
          <div className="modal__content">
            <ModalComponent />
          </div>
        </div>
      )
      : null;
  };
  Modal.propTypes = {
    component: PropTypes.string,
    onOverlayClick: PropTypes.func.isRequired
  };

  return Modal;
};


export default createModal;