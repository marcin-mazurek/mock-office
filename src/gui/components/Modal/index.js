export { default as createModal } from './Modal';
export { default as ModalConnect } from './ModalConnect';
