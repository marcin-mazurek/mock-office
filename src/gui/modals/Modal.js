import { connect } from 'react-redux';
import React from 'react';
import PropTypes from 'prop-types';
import AddServerModal from './AddServerModal';
import selectors from './selectors';

export const OVERLAY_CLICKED = 'modals/OVERLAY_CLICKED';

export const overlayClickedAction = () => ({
  type: OVERLAY_CLICKED
});

export const types = {
  addServerModal: AddServerModal,
};

export const Modal = ({ component, onOverlayClick }) => {
  const ModalComponent = types[component];
  const isOpen = !!ModalComponent;

  return isOpen
    ? (
      <div
        className="modal"
      >
        <button className="modal__overlay" onClick={onOverlayClick} />
        <div className="modal__content">
          <ModalComponent />
        </div>
      </div>
    )
    : null;
};
Modal.propTypes = {
  component: PropTypes.string,
  onOverlayClick: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  component: selectors.modalSelector(state)
});
const mapDispatchToProps = {
  onOverlayClick: overlayClickedAction
};
export const ModalConnect = connect(mapStateToProps, mapDispatchToProps)(Modal);
