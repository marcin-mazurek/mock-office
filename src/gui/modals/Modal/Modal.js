import React from 'react';
import PropTypes from 'prop-types';
import AddServerModal from './AddServerModal';

export const types = {
  addServerModal: AddServerModal,
};

export const OVERLAY_CLICKED = 'modals/OVERLAY_CLICKED';

export const overlayClickedAction = () => ({
  type: OVERLAY_CLICKED
});

export const Modal = ({ component, onOverlayClick }) => {
  const ModalComponent = types[component];
  const isOpen = !!ModalComponent;

  return isOpen
    ? (
      <div
        className="modal"
      >
        <button className="modal__overlay" onClick={onOverlayClick} />
        <div className="modal__content">
          <ModalComponent />
        </div>
      </div>
    )
    : null;
};
Modal.propTypes = {
  component: PropTypes.string,
  onOverlayClick: PropTypes.func.isRequired
};
