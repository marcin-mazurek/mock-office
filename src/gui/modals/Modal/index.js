export { default } from './Modal';
export { createModalConnect, OVERLAY_CLICKED, overlayClickedAction } from './ModalConnect';
export { default as AddServerModal } from './AddServerModal';
export { default as Modal } from './Modal';
