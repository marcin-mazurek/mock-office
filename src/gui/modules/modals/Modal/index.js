export { default } from './Modal';
export { createModalConnect, OVERLAY_CLICKED, overlayClickedAction } from './ModalConnect';
export { default as createModal } from './Modal';
