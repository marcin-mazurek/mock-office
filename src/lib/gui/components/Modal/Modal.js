import React from 'react';
import PropTypes from 'prop-types';
import { TransitionMotion, spring } from 'react-motion';

const createModal = (types) => {
  class Modal extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        items: []
      };
    }

    componentWillReceiveProps(props) {
      const { component } = props;
      const ModalComponent = types[component];
      const isOpen = !!ModalComponent;

      if (isOpen) {
        this.setState({
          items: this.state.items.concat([{ key: 'modal', style: { opacity: spring(1) } }])
        });
      } else {
        this.setState({
          items: []
        });
      }
    }

    willEnter() {
      return { opacity: 0 };
    }

    willLeave() {
      return { opacity: spring(0) };
    }

    render() {
      const { component, onOverlayClick } = this.props;
      const ModalComponent = types[component];
      const isOpen = !!ModalComponent;

      return (
        <TransitionMotion
          willEnter={this.willEnter}
          willLeave={this.willLeave}
          styles={this.state.items}
        >
          {
            interpolatingStyles =>
              <div>
                {
                  interpolatingStyles.map(config =>
                    <div
                      key={config.key}
                      className="modal"
                      style={config.style}
                    >
                      <button className="modal__overlay" onClick={onOverlayClick} />
                      <div className="modal__content">
                        {
                          isOpen ? <ModalComponent /> : null
                        }
                      </div>
                    </div>
                  )
                }
              </div>
          }
        </TransitionMotion>
      );
    }
  }

  Modal.propTypes = {
    component: PropTypes.string,
    onOverlayClick: PropTypes.func.isRequired
  };

  return Modal;
};

export default createModal;
