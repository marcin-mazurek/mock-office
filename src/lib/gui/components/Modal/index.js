import { createModalConnect } from 'modalo';
import AddServerModal from '../AddServerModal';
import { AddMockModalConnect } from '../AddMockModal';

const Modal = createModalConnect({
  AddServerModal,
  AddMockModal: AddMockModalConnect
});

export default Modal;
