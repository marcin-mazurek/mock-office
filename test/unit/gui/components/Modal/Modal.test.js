import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import R from 'ramda';
import { createModal } from '../../../../../src/gui/components/Modal';

describe('Modal', () => {
  test('no component snapshot', () => {
    const Modal = createModal({});
    const wrapper = shallow(
      <Modal onOverlayClick={() => {}} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('rendering addServerModal snapshot', () => {
    const TestModal = () => <div />;
    const Modal = createModal({ TestModal });
    const wrapper = shallow(
      <Modal component="TestModal" onOverlayClick={R.identity} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
