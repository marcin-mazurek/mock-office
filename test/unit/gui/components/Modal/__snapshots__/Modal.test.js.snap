// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal no component snapshot 1`] = `null`;

exports[`Modal rendering addServerModal snapshot 1`] = `
<div
  className="modal"
>
  <button
    className="modal__overlay"
    onClick={[Function]}
  />
  <div
    className="modal__content"
  >
    <TestModal />
  </div>
</div>
`;
